mu_repo/__init__.py

```diff
@@ -66,35 +66,39 @@ def CreateConfig(config_file='.mu_repo'):
 
     return Config.Create(contents)
 
+
 def CreateParams(args, config_file='.mu_repo'):
     return Params(CreateConfig(config_file), args, config_file)
 
 
-def SearchConfigFile(start_dir, recurse_limit=20):
+def SearchConfigFile(start_dir, name='.mu_repo', recurse_limit=20):
     """Search upwards in the file system starting in start_dir for a ".mu_repo" file, until the maximum limit or
     the root of the file-system is reached.
     """
-    start_dir = os.path.normpath(os.path.abspath(start_dir))
-    while True:
-        filename = os.path.join(start_dir, '.mu_repo')
+    start_dir = os.path.abspath(start_dir)
+    while recurse_limit >= 0:
+        filename = os.path.join(start_dir, name)
         if os.path.isfile(filename):
             return filename
 
-        recurse_limit -= 1
-        if recurse_limit == 0:
+        # give up searching if we are in a .git repository
+        if os.path.isdir(os.path.join(start_dir, '.git')):
             return None
```

Comment:
I think I didn't express myself properly here on the previous comment... It should work with this directory in this case. So, returning filename should work to make the chdir to the start dir -- or maybe this should return the directory instead of the full filename (so that it'd search for the .mu_repo locally in this dir later on -- in effect, this is how it works again anyways, so, maybe it'd be better having this as a SearchConfigFileDir returning the directory instead of returning the full path to the .mu_repo file).

What do you think?